examples/autokey_setup/main.tf

```
@@ -0,0 +1,24 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+module "autokey" {
+  source = "terraform-google-modules/terraform-google-kms/google//modules/autokey"
```

Apologies, bad copy/paste in my suggestion

```suggestion
  source = "terraform-google-modules/kms/google//modules/autokey"
```

---

There is some confusion here, we have updated the modules/autokey in this same PR and want to refer to the changed module in these example. 
IG only way to do that is refer actual source-dir in this PR. Once the updated modules/autokey is pushed to the tf registry use the resulting version here.

Please correct me if i missed something...

---

Hi @nb-goog - As part of the Lint and Int workflows we first run `module-swapper` which will automatically remove the version constraints and rewrite `source = "terraform-google-modules/kms/google//modules/autokey"` to `source = "../../modules/autokey"`.  That way the examples are always tested against the PR's contents.

Here is a similar example: https://github.com/terraform-google-modules/terraform-google-kms/actions/runs/12386236851/job/34573797497?pr=163#step:4:165

---

done, but havent included version this time (`version>~4.0`  ), i dont think module-swapper doesnt do anything regarding version. 

Also the behavior seems to be different when we run `make docker_test_lint` locally. 
IG model-swapper doesnt run there.

---

You can include a `version` for the example, `module-swapper` will comment it out when it switches to local.

---

done